deploy/quickstart/quick_start.py

```diff
@@ -0,0 +1,211 @@
+    docker_image = name + ":" + version
+    print("Pulling docker image " + docker_image)
+    call(["docker", "pull", docker_image], stdout=subprocess.DEVNULL)
+    return docker_image
+
+
+def request_input_from_list(message, options, default):
+    valid = False
+    provided = default
+    if not default in options:
+        printable_option_list = options.copy()
+        all_options = options.copy()
+        all_options = all_options + default
+    else:
+        all_options = options
+        printable_option_list = options.copy()
+        printable_option_list.remove(default)
+
+    printed_options_list = "(" + "/".join(str(e) for e in all_options) + ")"
+
+    while not valid:
+        provided = input(">> " + message + " " + printed_options_list + " [ " + default + " ] > ") or default
+        if provided in all_options:
+            valid = True
+        else:
+            print("Invalid option. Expected one of: " + printed_options_list + " but got: " + provided)
+            valid = False
+    print("Using: " + provided)
+    print()
+    return provided
+
+
+def make_or_replace_or_reuse_directory(path_to_make):
+    valid = False
+    time_to_exit = False
+    while not time_to_exit:
+        if path.isdir(path_to_make):
+            print("Directory " + path_to_make + " already exists. I can...")
+            print("1. Default: Re-use the existing directory which might leave behind some old files")
+            print("2. Delete the entire directory and all of its contents - and then remake it, completely empty")
+            provided = input("Replace or re-use? (1/2) [ 1 ] > ") or "1"
+            if provided is 1:
+                valid = True
+                time_to_exit = True
+            elif provided is 2:
+                try:
+                    call(["rm", "-rf", provided])
+                    os.mkdir(provided)
+                    valid = True
+                    time_to_exit = True
+                except OSError as e:
+                    print("OS error: " + e)
+                    valid = False
+                    time_to_exit = True
+            else:
+                print("Invalid input: got " + provided + " but expected one of (1/2)")
+                valid = False
+                time_to_exit = False
+        else:
+            try:
+                os.mkdir(path_to_make)
+                valid = True
+            except OSError as e:
+                print("OS error: " + e)
+                valid = False
+                time_to_exit = False
+
+    if not valid:
+        print("Cannot continue. Exiting")
+        exit(1)
+
+
+def request_input_path(message, default):
+    valid = False
+    absolute_path = path.abspath(default)
+    while not valid:
+        provided = input(">> " + message + " [ " + absolute_path + " ] > ") or absolute_path
+        absolute_path = path.abspath(provided)
+        print("Using: " + provided)
+
+        make_or_replace_or_reuse_directory(provided)
+
```

Comment:
missing a `valid = True` here, which will cause infinite loop i think

Reply:
You probably want to assign the` return value` from `make_or_replace_or_reuse_directory()`(which currently doe not have any return values) to the local `valid` to True?